CodeObject.js

```diff
@@ -4,9 +4,22 @@ function CodeObject (id, updated) {
   this._updated = updated || function () {};
   this.id = id;
   this._data = Object.create(null);
+  this._data.name = id;
   this._data.code = '';
 }
 
+CodeObject.prototype.setName = function (newName) {
```

Comment:
I think you may have missed the point of the `setData` method. The `CodeObject` class is meant to be observable, that is, an instance will call its `updated` callback whenever one of its properties changes.

So you're missing out on the big advantage of treating `name` as just another property: there's already plumbing in place so that whenever you do `someCodeObject.setData({name: "new name"})`, messages will be sent to all the clients to update them. You shouldn't even need a separate `code rename` event, as the `code update` event should be enough to update any property.

Reply:
In fact, I just tried removing the while `code rename` handler from `app.js` and renaming still works on all clients when you click `go live` — but not when just exiting the name field, since that depends on the `code rename` message. But even that could in theory use a `code update` message — you don't event have to send the code again as all updates are incremental!